File: pkg/types/alertmanagertypes/jira_metadata.go
package alertmanagertypes

type JiraMetadataRequest struct {
	ConnectionID string `json:"connection_id"`
	Project      string `json:"project"`
	IssueType    string `json:"issue_type"`
}

type JiraAllowedValue struct {
	Label string `json:"label"`
	Value string `json:"value"`
}

type JiraFieldMetadata struct {
	ID             string             `json:"id"`
	Name           string             `json:"name"`
	Required       bool               `json:"required"`
	SchemaType     string             `json:"schema_type,omitempty"`
	SchemaItems    string             `json:"schema_items,omitempty"`
	SchemaSystem   string             `json:"schema_system,omitempty"`
	SchemaCustom   string             `json:"schema_custom,omitempty"`
	SchemaCustomID int                `json:"schema_custom_id,omitempty"`
	AllowedValues  []JiraAllowedValue `json:"allowed_values,omitempty"`
}

type JiraMetadataResponse struct {
	Project   string              `json:"project"`
	IssueType string              `json:"issue_type"`
	Fields    []JiraFieldMetadata `json:"fields"`
}

File: frontend/src/types/api/channels/jiraMetadata.ts
export interface JiraMetadataRequest {
	connection_id: string;
	project: string;
	issue_type: string;
}

export interface JiraAllowedValue {
	label: string;
	value: string;
}

export interface JiraFieldMetadata {
	id: string;
	name: string;
	required: boolean;
	schema_type?: string;
	schema_items?: string;
	schema_system?: string;
	schema_custom?: string;
	schema_custom_id?: number;
	allowed_values?: JiraAllowedValue[];
}

export interface JiraMetadataResponse {
	project: string;
	issue_type: string;
	fields: JiraFieldMetadata[];
}
